include/tester/core/Application.hpp

```diff
@@ -1,18 +1,74 @@
 #pragma once
 
-#include<iostream>
+#include <iostream>
+#include <functional>
+#include <memory>
 
-#include"../db/Database.hpp"
-#include"../ssp/SoftphoneManager.hpp"
-#include"../vtcp/VTCPServer.hpp"
+#include "bsoncxx/builder/stream/document.hpp"
+
+#include "../db/Database.hpp"
+#include "../vtcp/VTCPServer.hpp"
+#include "../ssp/SoftphoneManager.hpp"
+#include "../db/Logger.hpp"
 
 class Application
 {
-private:
-    Database *_db;
-    SoftphoneManager *_sm;
-    VTCPServer *_vs;
 public:
-    Application(/* args */);
-    ~Application();
+    static Application& getInstance()
+    {
+        static Application instance;
+        return instance;
+    }
+
+    ~Application() = default;
+
+    void init(int server_port, int sm_port, const std::string& sm_domain, int sm_log_level, const std::string& db_domain)
+    {
+        _sm = std::make_shared<SoftphoneManager>(sm_port, sm_domain);
+        _vs = std::make_shared<VTCPServer>(server_port,
+            std::bind(&Application::startAutoTest, this, std::placeholders::_1),
+            std::bind(&Application::startManualTest, this),
+            std::bind(&Application::getHistoryHeaders, this),
+            std::bind(&Application::getHistoryLog, this, std::placeholders::_1));
+        _db = std::make_shared<Database>(db_domain);
+        _sm->pjLibraryInit(sm_log_level);
+    }
+
+    void run()
+    {
+        _vs->start();
+    }
+
+private:
+    Application()
+    {
+
+    }
+
+    bsoncxx::document::value startAutoTest(int amount)
+    {
+        auto log = _sm -> runSpamTest(amount);
+        _db->save("Test-Logs", log);
+        return log;
+    }
+
+    void startManualTest()
+    {
+
+    }
+
+    bsoncxx::document::value getHistoryHeaders()
+    {
+        return(_db -> getAllHeaders("Test-Logs"));
```

Comment:
remove the `return` parentheses and spaces on each end of the `->` operator

Reply:
done